### DISCUSSION TITLE
Self-hosted instance returning `Page Not Found` for all badges

### DISCUSSION THREAD

**Discussion by oe-josh-martin** [badges/shields#9521] [Q&A] [answered]
👋🏽

We've set up a self-hosted instance of Shields on Vercel, however, all badges are returning `Page Not Found` rather than the badge SVG. We're building from source and Shields instance stands up correctly on the Vercel URL, but it doesn't let us link out to the badges.

We've stood it up locally and it works fine, so unsure as to where the differences might be.

Thanks!

> Note: I've removed the package name and scope but you can still see the resulting execution code 

**Localhost**
<img width="554" alt="Screenshot 2023-08-29 at 12 34 32" src="https://github.com/badges/shields/assets/128401009/40ca843b-555e-4d7a-aaaf-1d7b2fdf7e33">

**Vercel**
<img width="565" alt="Screenshot 2023-08-29 at 12 35 51" src="https://github.com/badges/shields/assets/128401009/e7997003-e584-4b94-a9ec-6e7edd95248c">

**Comment by chris48s** [comment, marked as answer]
I've never tried running it on vercel, so I can't really guide you through this too much, but my guess is you're telling vercel to build the frontend and run that as a static site, but you are never starting the badge server. That would be why you've got a working "website" bit of the app but its not serving any actual images. I'm not really sure what steps you'd take on Vercel but basically you want to build the frontend with `npm run build` but then start the badge server (which can serve the built frontend as well as the badges) using either `node server` or `npm run start:server:prod`

**Comment by oe-josh-martin** [comment]
Ah of course, Vercel is only for static deployments not running servers isn't it 🙃. Thanks for the speedy reply @chris48s, will have to consider a different environment.
